aten/src/ATen/native/cpu/zmath.h

```diff
@@ -199,32 +200,56 @@ inline std::complex<double> trunc_impl (std::complex<double> z) {
 
 template <typename TYPE>
 inline TYPE max_impl (TYPE a, TYPE b) {
-  return std::max(a, b);
+  if (_isnan<TYPE>(a) || _isnan<TYPE>(b)) {
```

Comment:
Why not use `std::isnan`?

Reply:
@xuhdev ,  https://github.com/pytorch/pytorch/blob/9f7708eecb25268702ac107471fd13972fb4f5d2/aten/src/ATen/NumericUtils.h#L16 say **std::isnan isn't performant to use on integral types**.  there can use **std::isnan** as @VitalyFedyunin suggestion if not care Nan case for int.

Reply:
I think you are right, according to https://en.cppreference.com/w/cpp/numeric/math/isnan :

> A set of overloads or a function template accepting the arg argument of any integral type. Equivalent to (2) (the argument is cast to double).